torchx/workspace/docker_workspace.py

```diff
@@ -119,7 +131,7 @@ def build_workspace_and_update_role(
                     f"failed to pull image {role.image}, falling back to local: {e}"
                 )
             log.info("Building workspace docker image (this may take a while)...")
-            image, _ = self._docker_client.images.build(
+            build_events = self._docker_api_client.build(
```

Comment:
Why can't we simply use the second value in the tuple that `self._docker_client.images.build()` returns? It's the same build_events Iterable, isn't it? Then we don't have to copy the logic to retrieve the image id.

Reply:
If so we don't seem to print `error` events

Reply:
if i recall correctly, the first call is blocking and returns all the events at the same time, the second isnt and returns a stream which we can then print/process